compiler-rt/lib/scudo/standalone/mem_map_fuchsia.cpp

```diff
@@ -227,7 +234,7 @@ bool ReservedMemoryFuchsia::createImpl(UNUSED uptr Addr, uptr Size,
   zx_status_t Status = _zx_vmar_map(_zx_vmar_root_self(), ZX_VM_ALLOW_FAULTS, 0,
                                     getPlaceholderVmo(), 0, Size, &Base);
   if (UNLIKELY(Status != ZX_OK)) {
-    if (Status != ZX_ERR_NO_MEMORY || !AllowNoMem)
+    if (!IsNoMemError(Status) || !AllowNoMem)
```

Comment:
It'd be nice to someday find a way to reduce the brainwork to parse this expression and make it more readable.

Reply:
I agree that it's a bit obscure in its current form. What about inverting the if condition like this?
```
if (AllowNoMem && IsNoMemError(Status)) {
  ... // cleanup actions, if any (in practice, only in mapImpl's second if)
  return false;
}
dieOnError(Status, ...);
```

I can prepare a CL if you like this idea!

Reply:
Yes - that's a great improvement!